src/libs/antares/study/area/links.cpp

```diff
@@ -776,30 +782,23 @@ bool AreaLinksSaveToFolder(const Area* area, const char* const folder)
     return true;
 }
 
-void AreaLinkRemove(AreaLink* l)
+void AreaLinkRemove(AreaLink* link)
 {
-    if (!l)
+    if (!link)
         return;
 
     // Asserts
-    assert(l->from);
-    assert(l->with);
-
-    Area* f = l->from;
-    if (f && !f->links.empty())
-    {
-        f->detachLinkFromID(l->with->id);
-        f->detachLinkFromID(l->from->id);
-    }
+    assert(link->from);
+    assert(link->with);
 
-    Area* t = l->with;
-    if (t && !t->links.empty())
+    Area* areaFrom = link->from;
+    if (areaFrom && !areaFrom->links.empty())
```

Comment:
Note that if a link is removed, we remove it from the origin area, and not from the extremity area